src/libsync/propagatorjobs.cpp

```diff
@@ -256,12 +303,65 @@ void PropagateLocalRename::start()
             return;
         }
 
+        auto targetParentFolderPath = std::filesystem::path{};
+        auto targetParentPermissions = std::filesystem::perms{};
+        try {
+            const auto newDirPath = std::filesystem::path{targetFile.toStdWString()};
+            Q_ASSERT(newDirPath.has_parent_path());
+            targetParentFolderPath = newDirPath.parent_path();
+            auto parentFolderStatus = std::filesystem::status(targetParentFolderPath);
+            targetParentPermissions = parentFolderStatus.permissions();
+            if ((targetParentPermissions & std::filesystem::perms::owner_write) != std::filesystem::perms::owner_write) {
+                std::filesystem::permissions(targetParentFolderPath, targetParentPermissions | std::filesystem::perms::owner_write, std::filesystem::perm_options::replace);
+                emit propagator()->touchedFile(QString::fromStdWString(targetParentFolderPath.wstring()));
+            }
+        }
+        catch (const std::filesystem::filesystem_error &e)
+        {
+            qCWarning(lcPropagateLocalRename) << "exception when checking parent folder access rights" << e.what() << e.path1().c_str() << e.path2().c_str();
+        }
+
+        auto originParentFolderPath = std::filesystem::path{};
+        auto originParentPermissions = std::filesystem::perms{};
+        try {
+            const auto newDirPath = std::filesystem::path{existingFile.toStdWString()};
+            Q_ASSERT(newDirPath.has_parent_path());
+            originParentFolderPath = newDirPath.parent_path();
+            auto parentFolderStatus = std::filesystem::status(originParentFolderPath);
+            originParentPermissions = parentFolderStatus.permissions();
+            if ((originParentPermissions & std::filesystem::perms::owner_write) != std::filesystem::perms::owner_write) {
+                std::filesystem::permissions(originParentFolderPath, originParentPermissions | std::filesystem::perms::owner_write, std::filesystem::perm_options::replace);
+                emit propagator()->touchedFile(QString::fromStdWString(originParentFolderPath.wstring()));
+            }
+        }
+        catch (const std::filesystem::filesystem_error &e)
+        {
+            qCWarning(lcPropagateLocalRename) << "exception when checking parent folder access rights" << e.what() << e.path1().c_str() << e.path2().c_str();
+        }
+
+        const auto restoreTargetPermissions = [this] (const auto &parentFolderPath, const auto &parentPermissions) {
+            try {
+                if ((parentPermissions & std::filesystem::perms::owner_write) != std::filesystem::perms::owner_write) {
```

Comment:
@mgallien so, this block is identical to the first change in this file, could you move it into a private method with a descriptive name?